### PR TITLE
Add Python package building and upload actions

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: build_sdist/action.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: 'Build sdist'
+description: 'Build a Python package source distribution.'
+
+
```
Comment (paddyroddy) [inline build_sdist/action.yml]:
```suggestion
```

File: build_sdist/action.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+
+    - name: Build SDist
```
Comment (paddyroddy) [inline build_sdist/action.yml]:
```suggestion
    - name: Build sdist
```

File: build_sdist/action.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      shell: bash
+      run: pipx run build --sdist
```
Comment (paddyroddy) [inline build_sdist/action.yml:11]: Not heard of `pipx` until now. Why use it here?

File: build_wheels/action.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: 'Build wheels'
+description: 'Build Python package wheels.'
+
+
```
Comment (paddyroddy) [inline build_wheels/action.yml]:
```suggestion
```

File: upload_pypi/action.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name: 'Upload packages to PyPi'
+description: 'Upload built packages to the Python Packaging Index (PyPi).'
+
+
```
Comment (paddyroddy) [inline upload_pypi/action.yml]:
```suggestion
```

**Review by paddyroddy — APPROVED** [review]
Looks good

File: build_sdist/action.yml (hunk 3 above)
Comment (dstansby) [reply build_sdist/action.yml:11]: It runs the build in an isolated environment

**Pull request merged**